src/main/java/org/broadinstitute/hellbender/tools/funcotator/dataSources/DataSourceUtils.java

```diff
@@ -733,4 +734,17 @@ public static int datasourceComparator(final DataSourceFuncotationFactory f1, fi
             }
         }
     }
+
+    /**
+     * @param mainSourceFilePath  The main source file for this datasource.  Typically, specified in the config file as
+     *   {@link DataSourceUtils#CONFIG_FILE_FIELD_NAME_SRC_FILE}.  Never {@code null}
+     * @param name The name of this datasource.  Never {@code null}
+     * @return a new instance of {@link FeatureInput} that can be used for querying.
+     */
+    @VisibleForTesting
+    public static FeatureInput<? extends Feature> createFeatureInput(final String mainSourceFilePath, final String name) {
+        Utils.nonNull(mainSourceFilePath);
+        Utils.nonNull(name);
+        return new FeatureInput<>(mainSourceFilePath, name, Collections.emptyMap());
```

Comment:
You're calling a constructor that's `@VisibleForTesting` in a non-testing context. I'd like to suggest the following:

1. Pass the tool instance into `DataSourceUtils.createDataSourceFuncotationFactoriesForDataSources()`
2. Get your FeatureInputs by calling the *pre-existing* overload of `GATKTool.addFeatureInputsAfterInitialization()` that returns a `FeatureInput`
3. This will have the nice side effect of registering your FeatureInputs with the engine immediately.

This approach will allow you to eliminate the block of code in `Funcotator` proper that calls `addFeatureInputsAfterInitialization()`, and it will also make me happy by centralizing construction of the FeatureInputs in the engine. Win-win?

Reply:
Note that this suggestion would require making `GATKTool.addFeatureInputsAfterInitialization()` public rather than protected, which I think would be fine. If you're privileged enough to have an actual instance of a `GATKTool`, it makes sense that you'd be privileged enough to add extra inputs to it, run the tool, etc.

Reply:
Sounds good, though this will make testing somewhat more complicated, since I won't have a tool with which I can create features.

I may need to make a method that's `VisibleForTesting` that duplicates the old code.